src/OmegaPoker.t.sol

```diff
@@ -0,0 +1,181 @@
+}
+
+interface OsmLike {
+    function pass() external view returns (bool);
+}
+
+
+contract OmegaPokerTest is DSTest {
+    SpellLike    constant spell     = SpellLike(address(0));
+    SpellLike    constant prevSpell = SpellLike(address(0));
+
+    ChainLogLike constant changelog = ChainLogLike(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
+
+    PauseLike pause;
+    ChiefLike chief;
+    TokenLike govToken;
+
+    Hevm hevm;
+
+    OmegaPoker omegaPoker;
+
+    bytes20 constant CHEAT_CODE = bytes20(uint160(uint256(keccak256('hevm cheat code'))));
+
+    function setUp() public {
+        hevm = Hevm(address(CHEAT_CODE));
+        omegaPoker = new OmegaPoker();
+        omegaPoker.refresh();
+        pause = PauseLike(changelog.getAddress("MCD_PAUSE"));
+        chief = ChiefLike(changelog.getAddress("MCD_ADM"));
+        govToken = TokenLike(changelog.getAddress("MCD_GOV"));
+        hevm.warp(now + 3600);
+    }
+
+    function vote(SpellLike spell_) private {
+        if (chief.hat() != address(spell_)) {
+            hevm.store(
+                address(govToken),
+                keccak256(abi.encode(address(this), uint256(1))),
+                bytes32(uint256(999999999999 ether))
+            );
+            govToken.approve(address(chief), uint256(-1));
+            chief.lock(999999999999 ether);
+
+            assertTrue(!spell_.done());
+
+            address[] memory yays = new address[](1);
+            yays[0] = address(spell_);
+
+            chief.vote(yays);
+            chief.lift(address(spell_));
+        }
+        assertEq(chief.hat(), address(spell_));
+    }
+
+    function schedule(SpellLike spell_) public {
+        if (spell_.eta() == 0) {
+            spell_.schedule();
+        }
+        assertTrue(spell.eta() > 0);
+    }
+
+    function waitAndCast(SpellLike spell_) public {
+        uint256 castTime = now + pause.delay();
+
+        uint256 day = (castTime / 1 days + 3) % 7;
+        if(day >= 5) {
+            castTime += 7 days - day * 86400;
+        }
+
+        uint256 hour = castTime / 1 hours % 24;
+        if (hour >= 21) {
+            castTime += 24 hours - hour * 3600 + 14 hours;
+        } else if (hour < 14) {
+            castTime += 14 hours - hour * 3600;
+        }
+
+        hevm.warp(castTime);
+        spell_.cast();
+    }
+
+    function try_poke() internal returns (bool ok) {
+        (ok,) = address(omegaPoker).call(abi.encodeWithSignature("poke()"));
+    }
+
+    function test_poke() public {
+        if (address(prevSpell) != address(0) && !prevSpell.done()) {
+            vote(prevSpell);
+            schedule(prevSpell);
+            waitAndCast(prevSpell);
+        }
+        if (address(spell) != address(0) && !spell.done()) {
+            vote(spell);
+            schedule(spell);
+            assertTrue(!try_poke());
+            waitAndCast(spell);
+        }
+
+        assertTrue(try_poke());
+    }
+
+    function testRefresh() public {
+        omegaPoker.refresh();
+
```

Comment:
Should this have some assertions? Or at least a comment suggesting to add custom assertions based on the spell?

At the very least, should probably assert that the number of ilks tracked increases.

Reply:
Done